centreon/www/include/configuration/configObject/service/DB-Func.php

```diff
@@ -2919,23 +2919,22 @@ function updateServiceNotifOptionInterval_MC($service_id = null)
 
 function updateServiceNotifOptionTimeperiod($service_id = null, $ret = array())
 {
-    global $form, $pearDB;
+    global $pearDB;
 
-    if (!$service_id) {
+    if (! $service_id) {
```

Comment:
If $service_id is an integer, so the condition will be `! (int) $service_id > 0`  :)

Reply:
This method (updateServiceNotifOptionTimeperiod) always seems to be called with the serviceId argument, so you can change the signature to: 
updateServiceNotifOptionTimeperiod(int $service_id, $ret = []);
Then, each time you call the method, you need to cast the variable containing the service ID.
And finally, you can delete this condition.